### PR TITLE
Add basic Dixon & Coles model

### PR DESCRIPTION
**Pull request by anguswilliams91**
This PR implements a simple Dixon and Coles model (i.e., the original Dixon & Coles model with minimal modifications to make it a hierarchical model - meaning that there are independent priors for the attack and defence parameters).

I think we can use this as a base for building more complicated versions (including at least the one that is in the current version of `bpl`). There are a few things to add to the model here in order to reproduce the current one in bpl:

* Team-level covariates in the prior.
* Per-team home advantage.
* Correlation between the attack and defence parameters in the prior.

I'd also be keen to hear thoughts on some API changes from the other version of `bpl`:

* I removed the dependency on `pandas` - stuff is just passed around in dicts or lists. It seemed a heavy dependency but it's not really required (the `.dict()` method could be used on dataframes if the upstream data processing uses `pandas`).
* the training data is now passed to `fit` (a la scikit-learn), instead of init.
* the two main functions for prediction are `predict_score_proba` and `predict_outcome_proba` (both vectorised) which produce predictions for specific scorelines and overall outcomes (home win, etc.) respectively.

### CODE REVIEW

**Review by nbarlowATI — APPROVED** [review]
Looks good!    API looks fine from an AIrsenal perspective.
Dropping the pandas dependency makes sense (though i can confirm the DixonColesMatchPredictor works just fine if you give it a pandas DataFrame in fit() ).

Not sure about how poetry.lock is supposed to work - I got lots of errors when I naively tried to just do `poetry install`,  and ended up `pip install`ing loads of packages by hand.   Probably I was just misunderstanding poetry, but a couple of lines in the README would be good in a future version.

Definitely good to merge this now though!
